realm/realm-library/src/androidTest/java/io/realm/DynamicRealmObjectTests.java

```diff
@@ -1727,4 +1727,23 @@ public void run() {
         thread.start();
         TestHelper.awaitOrFail(threadFinished);
     }
+
+    @Test
+    public void getNullableBoolean() {
+        realm.executeTransaction(realm -> {
+            AllJavaTypes object = realm.createObject(AllJavaTypes.class, 1000L);
+            object.setFieldBoolean(null);
+
+            assertNull(object.isFieldBoolean());
+        });
+        realm.close();
+
+        DynamicRealm dynamicRealm = DynamicRealm.getInstance(realm.configuration);
+        dynamicRealm.refresh();
```

Comment:
You don't need to call refresh. It should be up to date when opened.

Reply:
It does not work without it. Maybe symptom of another issue. The below sequence runs without asserting
```
DynamicRealm dynamicRealm = DynamicRealm.getInstance(realm.configuration);
assertEquals(1, dynamicRealm.where("AllJavaTypes").count());
dynamicRealm.refresh();
assertEquals(2, dynamicRealm.where("AllJavaTypes").count());
```

Reply:
Hmm, I don't like that. The underlying read transactions should not be shared between `Realm` and `DynamicRealm`. It indicates a bigger problem we need to look into. Probably some caching going wrong somewhere 😕

Reply:
Yes. A dynamic realm with the same configuration was already setup. Sorry.